src/pm/hydra/ui/mpich/utils.c

```diff
@@ -449,6 +449,13 @@ static void prepend_rank_help_fn(void)
 static HYD_status prepend_rank_fn(char *arg, char ***argv)
 {
     HYD_status status = HYD_SUCCESS;
+    int val;
+
+    if (MPL_env2int("MPIEXEC_PREFIX_DEFAULT", &val)) {
+        if (val >= 1) {
+            goto fn_exit;
+        }
+    }
```

Comment:
The point of processing environment in `check_envs` is so we don't check environment again in another place.